### PR TITLE
DOCS-#4466: Recommend GitHub issues over bug_reports@modin.org.

### LINKED ISSUE #4466: Replace bug_reports@modin.org with a link to create a new issue on GitHub
- **Modin version**: cf0eb393daa41abdd8cdf32b52ceee938cdcbe1a

### Describe the problem
There are a few links to bug_reports@modin.org in the source, e.g. [here](https://github.com/modin-project/modin/blob/6d419655bba85b3b0f2b827624788f62d7c729cd/modin/error_message.py#L62). We should replace those with a link to the GitHub issues and a message saying to search the existing GitHub issues and file a new one if the bug hasn't already been reported.

### PR DESCRIPTION
**Pull request by mvashishtha**
Signed-off-by: mvashishtha <mahesh@ponder.io>

## What do these changes do?

- [x] commit message follows format outlined [here](https://modin.readthedocs.io/en/latest/development/contributing.html#commit-message-formatting)
- [x] passes `flake8 modin/ asv_bench/benchmarks scripts/doc_checker.py`
- [x] passes `black --check modin/ asv_bench/benchmarks scripts/doc_checker.py`
- [x] signed commit with `git commit -s` 
- [x] Resolves #4466
- [x] tests added and passing
- [x] module layout described at `docs/development/architecture.rst` is up-to-date 
- [x] added (Issue Number: PR title (PR Number)) and github username to release notes for next major release

### CODE REVIEW

**Comment by anmyachev** [comment]
@mvashishtha need to resolve conflict

**Comment by mvashishtha** [comment]
> @mvashishtha need to resolve conflict

@anmyachev Done.

**Comment by YarShev** [comment]
Should we also recommend GitHub issues over feature_requests@modin.org? Probably, as part of a separate issue.

**Review by devin-petersohn — COMMENTED** [review]
The messages are detailed, but long. Is there a way to make them shorter? Maybe we can remove the clause about checking that the issue already exists?

**Comment by mvashishtha** [comment]
> Should we also recommend GitHub issues over [feature_requests@modin.org](mailto:feature_requests@modin.org)? Probably, as part of a separate issue.

I filed #4487 for that.

**Comment by mvashishtha** [comment]
> The messages are detailed, but long. Is there a way to make them shorter? Maybe we can remove the clause about checking that the issue already exists?

@devin-petersohn I removed the clauses about checking that the issue already exists.

File: modin/core/io/text/excel_dispatcher.py [hunk 1]
```diff
@@ -73,8 +73,10 @@ def _read(cls, io, **kwargs):
             return cls.single_worker_read(io, **kwargs)
 
         warnings.warn(
-            "Parallel `read_excel` is a new feature! Please email "
-            + "bug_reports@modin.org if you run into any problems."
+            "Parallel `read_excel` is a new feature! If you run into any "
+            + "problems, please visit https://github.com/modin-project/modin/issues. "
+            + "If you find a new issue and can't file it on GitHub, please "
```
Comment (YarShev) [inline modin/core/io/text/excel_dispatcher.py:78]:
```suggestion
            + "If you find a new issue or can't file it on GitHub, please "
```

**Review by devin-petersohn — APPROVED** [review]

**Pull request merged**